packages/wrangler/src/cloudchamber/curl.ts

```diff
@@ -0,0 +1,153 @@
+import { randomUUID } from "crypto";
+import { bold, brandColor, cyanBright, yellow } from "@cloudflare/cli/colors";
+import { OpenAPI } from "./client";
+import { ApiError } from "./client/core/ApiError";
+import { request } from "./client/core/request";
+import type { Config } from "../config";
+import type {
+	CommonYargsOptionsJSON,
+	StrictYargsOptionsToInterfaceJSON,
+} from "../yargs-types";
+import type yargs from "yargs";
+
+export function yargsCurl(args: yargs.Argv<CommonYargsOptionsJSON>) {
+	return args
+		.positional("path", { type: "string", default: "/" })
+		.option("header", {
+			type: "array",
+			alias: "H",
+			describe: "Add headers in the form of --header <name>:<value>",
+		})
+		.option("data", {
+			type: "string",
+			describe: "Add a JSON body to the request",
+			alias: "D",
+		})
+		.option("method", {
+			type: "string",
+			alias: "X",
+			default: "GET",
+		})
+		.option("silent", {
+			describe: "Only output response",
+			type: "boolean",
+			alias: "s",
+		})
+		.option("verbose", {
+			describe: "Print everything, like request id, or headers",
+			type: "boolean",
+			alias: "v",
+		})
+		.option("use-stdin", {
+			describe: "Equivalent of using --data-binary @- in curl",
+			type: "boolean",
+			alias: "stdin",
+		});
+}
+
+export async function curlCommand(
+	args: StrictYargsOptionsToInterfaceJSON<typeof yargsCurl>,
+	config: Config
+) {
+	await requestFromCmd(args, config);
+}
+
+async function read(stream: NodeJS.ReadStream) {
+	const chunks = [];
+	for await (const chunk of stream) {
+		chunks.push(chunk);
+	}
+	return Buffer.concat(chunks).toString("utf8");
+}
+
+async function requestFromCmd(
+	args: {
+		path: string;
+		method: string;
+		header: (string | number)[] | undefined;
+		data?: string;
+		silent?: boolean;
+		verbose?: boolean;
```

Comment:
Do these have meaning over and above the `WRANGLER_LOG` env var that can control what `logger` methods actually write to the console?
Is there a reason you are not using the `logger`?

Reply:
I think we'd need to be careful with the output here. Is it expected that this command's output will be piped to other processes? If so, would changes to the output be considered a breaking change?

Since we've provided a --json flag, we should guide scripting users to that terse output which will reflect the backcompat of the api being hit. Leaving us more free to iterate on the human-readable output without fear of breaking anyone's workflow.

I don't see a problem with the `--verbose` flag, since `WRANGLER_LOG` or `--log-level` don't have a concept of verbosity, only debug logging. My only issue is if it encourages users to script against stdout, preventing us from iterating on this command.